admin-dev/themes/new-theme/js/pages/product/store/mutations.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import * as types from './mutation-types';
+
+/**
+ * Mutations for Vuex store
+ */
+export default {
+  [types.SET_PRICE_TAX_EXCLUDED](state, price) {
+    state.priceTaxExcluded = price;
```

Comment:
I don't understand why prices are in a VueX store, it's a not a state, it's a property / data in components.

Reply:
It's more like a property THEN a state inside a component, but if the price is used anywhere outside the component, using it in a store isn't a bad practice I think.

Remember, store should be handling states that are global, like an user account available everywhere in the app, if the price is global and will be manipulated anywhere but not only in one component, that should be fine

Reply:
We must be sure it needs to be used as a state like an user account, and not a way to store everything when properties are enough.

Reply:
That's the idea here, the price can be edited in two different tabs (Basic infos and Price tab) But the inputs need to be synced between those tabs So we need to integrate a way in a Symfony page to sync some inputs

Vuex seemed like a good candidate to use as a basis, as it already manages this kind of behaviour inside a Vue app, and since we tend to use Vue in the future at least the store part would already be based on Vuex

Keep in mind that THIS IS NOT a Vue app, it's a Symfony page, with form inputs, most of the logic will be done using jquery like ALL the other migrated pages. Vuex here is only used as a common global state for the page to make sure every input is up to date

Reply:
This is why it's also weird, loading a Vue instance for not using it only to have VueX loaded when you can have a simple `Object` management is not a good idea.

FYI:
Vuex checks for [the existence of Vue](https://github.com/vuejs/vuex/blob/v3.0.1/dist/vuex.js#L308).
Vuex [depends largely on Vue for its reactivity inner workings](https://github.com/vuejs/vuex/blob/v3.0.1/dist/vuex.js#L330).

https://github.com/vuejs/vuex/issues/1172#issuecomment-370692728